core/src/main/java/hudson/model/FingerprintCleanupThread.java

```diff
@@ -107,11 +107,13 @@ private void deleteIfEmpty(File dir) {
     private boolean check(File fingerprintFile, TaskListener listener) {
         try {
             Fingerprint fp = loadFingerprint(fingerprintFile);
-            if (fp == null || !fp.isAlive()) {
+            if (fp == null || (!fp.isAlive() && !fp.isFingerprintDeletionBlocked()) ) {
                 listener.getLogger().println("deleting obsolete " + fingerprintFile);
                 fingerprintFile.delete();
                 return true;
             } else {
+                if (!fp.isAlive() && fp.isFingerprintDeletionBlocked())
+                    listener.getLogger().println(("Facet blocked deletion of " + fingerprintFile));
```

Comment:
Would be great to reference the facet blocking it

Reply:
A facet by default IIUC only has timestamp and fingerprint attached with it. How would I reference the facet? Should I do it by timestamp? Or should i use the toString method?

Reply:
What about just printing a className? It would at least give the user some idea which facet blocks it

Reply:
@oleg-nenashev Done, I also added the timestamp of creation. Let me know if anything more is required